### PR TITLE
Study sessions

### PR DESCRIPTION
**Pull request by sciffany**
In this version, one is able to enter study mode using the command "study INDEX"

One is also able to exit from study mode using the command "done"

### CODE REVIEW

File: src/main/java/seedu/address/logic/StudyView.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import javafx.collections.ListChangeListener;
+import javafx.collections.transformation.FilteredList;
+import seedu.address.logic.commands.DoneCommand;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+import seedu.address.model.deck.Card;
+import seedu.address.model.deck.Deck;
+
+public class StudyView implements ListViewState {
```
Comment (xsot) [inline src/main/java/seedu/address/logic/StudyView.java:15]: `ListViewState` is meant to be for a `ViewState` that has a list interface.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -228,6 +228,21 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+//tag::studysessions[]
+=== Study Sessions feature
+==== Current Implementation
+
+The third ViewState is called `StudyState`.
+
+This state can be accessed when user executes the command `select DECK_NUMBER` and then `study`.
```
Comment (xsot) [inline docs/DeveloperGuide.adoc:238]: Invocation is inconsistent with PR.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -73,6 +74,15 @@ public void changeDeck(Deck deck) {
         setSelectedItem(null);
     }
 
+    public void studyDeck(Deck deck) {
+        viewState = new StudyView(this, deck);
+        // TODO: change this to above after migrating global cards list
```
Comment (xsot) [inline src/main/java/seedu/address/model/ModelManager.java:79]: This comment is copied out of context

File: src/main/java/seedu/address/ui/UiPart.java [hunk 4]
```diff
@@ -79,7 +79,6 @@ private void loadFxmlFile(URL location, T root) {
      * Returns the FXML file URL for the specified FXML file name within {@link #FXML_FILE_FOLDER}.
      */
     private static URL getFxmlFileUrl(String fxmlFileName) {
-        requireNonNull(fxmlFileName);
```
Comment (xsot) [inline src/main/java/seedu/address/ui/UiPart.java]: Why was this removed?

File: src/main/java/seedu/address/logic/StudyView.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    private final Model model;
+    public final FilteredList<Card> filteredCards;
+    private final SimpleObjectProperty<Card> selectedCard = new SimpleObjectProperty<>();
+    private final Deck activeDeck;
+
+    public StudyView(Model model, Deck deck) {
+        this.model = model;
+        this.activeDeck = deck;
+        filteredCards = new FilteredList<>(deck.getCards().asUnmodifiableObservableList());
```
Comment (xsot) [inline src/main/java/seedu/address/logic/StudyView.java:24]: Should not be a filtered list, unless you actually want to be able to filter the list.

File: src/main/java/seedu/address/logic/StudyView.java (hunk 1 above)
Comment (sciffany) [reply src/main/java/seedu/address/logic/StudyView.java:15]: yep, will change all these once I manage to create a UI that works to display a single item

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (sciffany) [reply docs/DeveloperGuide.adoc:238]: thanks!

File: src/main/java/seedu/address/ui/UiPart.java (hunk 4 above)
Comment (sciffany) [reply src/main/java/seedu/address/ui/UiPart.java]: shoot thanks for noticing, returning it now

File: src/main/java/seedu/address/model/ModelManager.java (hunk 3 above)
Comment (sciffany) [reply src/main/java/seedu/address/model/ModelManager.java:79]: Alright, will take note to remove this comment when I rewrite study deck in the next PR

File: src/main/java/seedu/address/logic/StudyView.java (hunk 5 above)
Comment (sciffany) [reply src/main/java/seedu/address/logic/StudyView.java:24]: as of now I'm only using filtered list because I want to display whatever we can display.

File: src/main/java/seedu/address/logic/StudyView.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public Deck getActiveDeck() {
+        return activeDeck;
+    }
+
+    /**
+     * Ensures {@code selectedItem} is a valid card in {@code filteredItems}.
+     */
+    private void ensureSelectedItemIsValid(ListChangeListener.Change<? extends Card> change) {
```
Comment (ChangDarren) [inline src/main/java/seedu/address/logic/StudyView.java:45]: This method should be gone eventually when the new UI interface for study session is created.
